ansible_rulebook/rule_set_runner.py

```diff
@@ -208,13 +233,8 @@ async def _drain_actionplan_queue(self):
                     )
                     self.active_actions.add(task)
                     task.add_done_callback(self.active_actions.discard)
```

Comment:
It is not a bug, but will the logic slightly more readable to remove this line and delete the task from active_actions inside the new callback?

Reply:
@bzwei They have to be distinct because in the other call back I am checking to see if the active_actions is empty to cancel the action plan task.

Reply:
@mkanoor I meant you could do `active_actions.discard(task)` in the new callback.

Reply:
@bzwei Updated.